StanwoodCore/Core/Storage.swift

```diff
@@ -49,32 +56,41 @@ extension Stanwood {
          Directory to save data
          */
         public enum Directory {
+            
             /// Only documents and other data that is user-generated, should be stored in the Documents directory and will be automatically backed up by iCloud.
-            case documents
+            case documents(customDirectory: String?)
             
             /// Data that can be downloaded again or regenerated should be stored in the Caches directory.
-            case caches
+            case caches(customDirectory: String?)
+            
+            /// Data that is user-generated should be stored in the Library directory.
+            case library(customDirectory: String?)
         }
         
         /// Returns URL constructed from specified directory
-        static fileprivate func getURL(for directory: Directory) -> URL {
+        static fileprivate func getURLComponents(for directory: Directory) throws -> (url: URL, isCustomDirectory: Bool) {
             var searchPathDirectory: FileManager.SearchPathDirectory
+            var customDirectory: String?
             
             switch directory {
-            case .documents:
+            case .documents(customDirectory: let directory):
+                customDirectory = directory
                 searchPathDirectory = .documentDirectory
-            case .caches:
+            case .caches(customDirectory: let directory):
+                customDirectory = directory
                 searchPathDirectory = .cachesDirectory
+            case .library(customDirectory: let directory):
```

Comment:
Should probably point this at the library subdirectories https://developer.apple.com/library/archive/documentation/FileManagement/Conceptual/FileSystemProgrammingGuide/FileSystemOverview/FileSystemOverview.html#//apple_ref/doc/uid/TP40010672-CH2-SW1

Reply:
Fixed